src/commands/key.rs

```diff
@@ -159,8 +162,7 @@ impl KeyDeleteCommand {
             .with_prompt("are you sure you want to delete your signing key")
             .interact()?
         {
-            // delete_signing_key(&self.common.registry, config)?;
-            delete_signing_key(&self.common.registry, config.keys.clone().expect("Please set a default signing key by typing `warg key set <alg:base64>` or `warg key new"), config.home_url.clone())?;
+            delete_signing_key(self.common.registry.as_deref(), config.keys.as_ref().expect("Please set a default signing key by typing `warg key set <alg:base64>` or `warg key new"), config.home_url.as_deref())?;
```

Comment:
Same comment as above (shouldn't use `expect` here):

```suggestion
            delete_signing_key(self.common.registry.as_deref(), config.keys.as_ref().context("please set a default signing key by typing `warg key set <alg:base64>` or `warg key new")?, config.home_url.as_deref())?;
```